plugins/modules/aci_esg.py

```diff
@@ -0,0 +1,375 @@
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import (
+    ACIModule,
+    aci_argument_spec,
+)
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        ap=dict(type="str", aliases=["app_profile", "app_profile_name"]),
+        esg=dict(type="str", aliases=["name", "esg_name"]),
+        admin_state=dict(type="str", choices=["no", "yes"]),  # ESG Admin State
+        vrf=dict(type="str", aliases=["vrf_name"]),  # ESG VRF name
+        description=dict(type="str", aliases=["descr"]),
+        intra_esg_isolation=dict(
+            type="str",
+            choices=["enforced", "unenforced"],
+        ),  # Intra ESG Isolation
+        preferred_group_member=dict(type="str", choices=["exclude", "include"]),  # Preferred Group Member
+        state=dict(
+            type="str",
+            default="present",
+            choices=["absent", "present", "query"],
+        ),
+        name_alias=dict(type="str"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "ap", "esg"]],
+            ["state", "present", ["tenant", "ap", "esg"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+    tenant = module.params.get("tenant")
+    ap = module.params.get("ap")
+    esg = module.params.get("esg")
+    admin_state = module.params.get("admin_state")
+    vrf = module.params.get("vrf")
+    description = module.params.get("description")
+    intra_esg_isolation = module.params.get("intra_esg_isolation")
+    preferred_group_member = module.params.get("preferred_group_member")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+
+    # VRF Selection - fvRsScope
+    child_configs = [dict(fvRsScope=dict(attributes=dict(tnFvCtxName=vrf)))]
```

Comment:
I would move this into the if state == "present section" as in the other cases we don't need it.